backend/de.metas.business/src/main/java/org/adempiere/mm/attributes/api/impl/AttributeDAO.java

```diff
@@ -903,6 +904,14 @@ public boolean nullSafeASIEquals(
 		return firstAttributeSet.equals(secondAttributeSet);
 	}
 
+	@Override
+	public boolean isHUUniqueAttribute(@NonNull final String trxName, final int m_product_id, @NonNull final I_M_Attribute attribute)
+	{
+		final String sql = "SELECT 1 FROM M_HU_UniqueAttribute WHERE m_product_id=? AND m_attribute_id=? AND value=?";
```

Comment:
Why are you not using queryBuilder? 
Also, please add the new method in the HUUniqueAttributesRepository: https://github.com/metasfresh/metasfresh/blob/master/backend/de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/attribute/impl/HUUniqueAttributesRepository.java

Reply:
and
* use thread inherited trx
* ProductId
* AttributeId

Reply:
and to match u could use QueryBuilder as @metas-rc  said and then org.compiere.model.IQuery#anyMatch